src/ServiceControl.Audit/Auditing/AuditPersister.cs

```diff
@@ -53,7 +54,7 @@ public async Task<IReadOnlyList<MessageContext>> Persist(IReadOnlyList<MessageCo
             {
 
                 // deliberately not using the using statement because we dispose async explicitly
-                unitOfWork = await unitOfWorkFactory.StartNew(contexts.Count);
+                unitOfWork = await unitOfWorkFactory.StartNew(contexts.Count, cancellationToken);
```

Comment:
Should we pass the token to Task.When all as well?

Reply:
Should we also pass the token to RecordProcessedMessage, RecordSagaSnapshot and RecordKnownEndpoint?

Reply:
> Task.When

Task.WhenAll does not have a CT arg.

Reply:
> Should we also pass the token to RecordProcessedMessage, RecordSagaSnapshot and RecordKnownEndpoint?

Added in latest commit